### DISCUSSION TITLE
Gap-filled NDVI timeseries to xarray

### DISCUSSION THREAD

**Discussion by singhamninder** [aazuspan/wxee#51] [Q&A] [answered]
Hi, 
Thanks for putting together this excellent package. I find it very useful for a lot of my workflows.
I have exported an ImageCollection to assets, consisting of gap-filled smooth NDVI time-series following this [tutorial](https://courses.spatialthoughts.com/end-to-end-gee.html#savitzky-golay-smoothing). Then I am using the following code to create a xarray - 
```
chico_ts = wxee.TimeSeries("users/amninder/chico_ic") \
            .sort("system:time_start") \
            .filterDate("2018-01", "2018-03") \
            .select(["smoothed"])

ts_arr = chico_ts.wx.to_xarray(region=polygon)
```
However, I keep getting nan values for data variables. Any thoughts on how I can fix that?

**Comment by aazuspan** [comment, marked as answer]
Hi @singhamninder, glad you're finding wxee useful!

Are you sure the NaN pixels you're getting aren't masked in Earth Engine? `to_xarray` assigns NaNs to masked pixels, so maybe something is going wrong with the gap filling and those pixels have no data. 

If you haven't already, I would try checking those images without `wxee`, either by looking at them with [geemap](https://github.com/giswqs/geemap) or downloading them manually with `ee.batch.Export.image.toDrive`. If those pixels *do* have valid data and it gets lost by `to_xarray`, I'll definitely look into that!

Also, if you want to post your entire code (and make the assets public) I can take a closer look.

EDIT: Adding the answer from below to the top:

> Okay, you just need to set the scale parameter. This should work!
> ```python
> ts_arr = chico_ts.wx.to_xarray(region=polygon, scale=10)
> ```
> If you don't specify a scale, Earth Engine tries to guess based on the image projection. It definitely got it wrong here, which is why it just gave you 1 pixel with no data.

**Reply by singhamninder** [reply]
Dear @aazuspan, thank you for your prompt response! Here is the script I have used for gap-filling and smoothing the time series https://code.earthengine.google.com/00638f1ee80267c14831d7718e736a72. And here is the ImageCollection that I have exported https://code.earthengine.google.com/?asset=users/amninder/chico_ic. 
I checked the images using `geemap` and the pixels do have data (as can be seen in the photo below). I am new to using xarrays so not sure if I am doing everything right, but would appreciate any help!

```
import geemap
import geemap.colormaps as cm
Map = geemap.Map(center=(40, -100), zoom=4, basemap="HYBRID")

chico_ic = ee.ImageCollection("users/amninder/chico_ic") \
            .sort("system:time_start") \

Map.addLayer(chico_ic.toBands(),{}, 'NDVI')
Map.addLayer(chico_ic.first(),
            {'min': 0, 'max': 1, 'palette': cm.palettes.ndvi},
            'first')
Map
```
![Screen Shot 2022-07-05 at 7 01 16 PM](https://user-images.githubusercontent.com/91866865/177452800-fc423d48-ebe4-40bf-a062-4d9521227e99.png)

**Reply by aazuspan** [reply]
Thanks for the details, and for confirming that the data isn't masked. I'd like to take a closer look and see if I can recreate the issue, but I'm not able to access the asset you linked. Could you make sure it's shared? If you click on the asset in the Code Editor there's a "Share" tab in the top right that you can set to "Anyone can read".

**Reply by singhamninder** [reply]
Oops sorry about that. Just shared it

**Reply by aazuspan** [reply]
Okay, you just need to set the `scale` parameter. This should work!

```python
ts_arr = chico_ts.wx.to_xarray(region=polygon, scale=10)
```

If you don't specify a scale, Earth Engine tries to guess based on the image projection. It definitely got it wrong here, which is why it just gave you 1 pixel with no data.

**Reply by singhamninder** [reply]
It works! Thank you so much for your time :)

**Reply by aazuspan** [reply]
No problem!
